### PR TITLE
Fix docstrings and documentation for asreview classifiers

### PR DESCRIPTION
**Pull request by J535D165**
This PR 

- Fixes docstrings for classifiers, balance strategies, feat extraction and query strategies. 
- Extends the reference docs https://asreview.readthedocs.io/en/latest/reference.html
- Add class members to docs
- Add models that depend on optional dependencies to docs
- Add models that depend on optional dependencies to submodules (`__init__.py`)
- Applies YAPF formatter to submodules. 
- Adds CI pipeline for bare minimum of dependencies.  
- Tests models for import errors if dependencies are not available.
- A few more things.

### CODE REVIEW

File: asreview/review/base.py [hunk 1]
```diff
@@ -259,7 +260,13 @@ def _stop_iter(self, query_i, n_pool):
         return stop_iter
 
     def n_pool(self):
```
Comment (J535D165) [inline asreview/review/base.py:262]: property? @qubixes

File: asreview/review/base.py [hunk 2]
```diff
@@ -460,7 +467,14 @@ def train(self):
             self.query_i_classified = 0
 
     def statistics(self):
```
Comment (J535D165) [inline asreview/review/base.py:469]: property? @qubixes

File: asreview/balance_strategies/double.py [hunk 3]
```diff
@@ -20,6 +20,11 @@
 from asreview.balance_strategies.simple import SimpleBalance
 from asreview.utils import get_random_state
 
+PARAM_A = 2.155
```
Comment (qubixes) [inline asreview/balance_strategies/double.py:23]: Why this restructuring?

Comment (J535D165) [reply asreview/balance_strategies/double.py]: Reverted.

**Pull request merged**